### PR TITLE
refactor(changelog): Relocate in-range release fetching

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/pr/changelog/index.ts [hunk 1]
```diff
@@ -48,7 +46,7 @@ export async function getChangeLogJSON(
 
     return res;
   } catch (err) /* istanbul ignore next */ {
-    logger.error({ config: args, err }, 'getChangeLogJSON error');
+    logger.error({ config: config, err }, 'getChangeLogJSON error');
```
Comment (zharinov) [inline lib/workers/repository/update/pr/changelog/index.ts:49]:
```suggestion
    logger.error({ config, err }, 'getChangeLogJSON error');
```

**Review by viceice — APPROVED** [review]
can you try to also decouple platform specific logic? we've a lot duplicate code for GitHub and gitlab.

**Review by viceice — APPROVED** [review]
can you try to also decouple platform specific logic? we've a lot duplicate code for GitHub and gitlab.

**Comment by zharinov** [comment]
Okay, will try in one of upcoming PRs...

**Comment by viceice** [comment]
> Okay, will try in one of upcoming PRs...

i like to add gitea support after that 🙃

**Pull request merged**